test/e2e/secrets_test.go

```diff
@@ -128,6 +129,31 @@ func testUserDataTamper(t *testing.T) {
 	}
 }
 
+// testPrivateKeyChange alters the private key used to SSH into instances and ensures nodes are updated properly
+func testPrivateKeyChange(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	// This test cannot be run on vSphere because this random key is not part of the vSphere template image.
+	// Moreover this test is platform agnostic so is not needed to be run for every supported platform.
+	if tc.CloudProvider.GetType() == config.VSpherePlatformType {
+		t.Skipf("Skipping for %s", config.VSpherePlatformType)
+	}
+	err = tc.createPrivateKeySecret(false)
+	require.NoError(t, err, "error replacing known private key secret with a random key")
+
+	// Ensure Machines nodes are re-created and configured using the new private key
+	err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, false, false, false)
+	assert.NoError(t, err, "error waiting for Windows nodes configured with newly created private key")
+	// Ensure public key hash and encrypted username annotations are updated correctly on BYOH nodes
+	err = tc.waitForWindowsNodes(gc.numberOfBYOHNodes, false, false, true)
+	assert.NoError(t, err, "error waiting for Windows nodes updated with newly created private key")
+
+	// Re-create the known private key so SSH connection can be re-established
+	// TODO: Remove dependency on this secret by rotating keys as part of https://issues.redhat.com/browse/WINC-655
+	require.NoError(t, tc.createPrivateKeySecret(true), "error confirming known private key secret exists")
```

Comment:
I'm still on the fence about this but it seems to be working...
This operation is going to bleed into the deletion suite as the WMCO will be handling this first and could taint the results that we expect when doing certain deletion actions. But what we are testing for in the BYOH reconfiguration, specifically the removal of binaries and services, shouldnt be triggered by this.

I'm fine with this going in but we need to handle the TODO story ASAP so that we dont open ourselves up to missing certain edge cases